### PR TITLE
Read file contents concurrently

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
This is now ready for review

File: cli/src/diff_util.rs [hunk 1]
```diff
@@ -677,19 +677,48 @@ fn show_unified_diff_hunks(
...
+) -> Pin<
+    Box<
+        dyn Stream<
+                Item = (
+                    RepoPathBuf,
+                    BackendResult<(MaterializedTreeValue, MaterializedTreeValue)>,
+                ),
+            > + 'a,
+    >,
+> {
```
Comment (yuja) [inline cli/src/diff_util.rs:692]: Better to return `impl Stream`? The complete type looks super complex.

Appears that `impl Stream<Item = (..)> + 'a` works, and `Box::pin()` can be removed.

File: cli/src/diff_util.rs [hunk 2]
```diff
@@ -677,19 +677,48 @@ fn show_unified_diff_hunks(
...
+                    Err(err) => (path, Err(err)),
+                    Ok((before, after)) => {
+                        let before_future = materialize_tree_value(store, &path, before);
+                        let after_future = materialize_tree_value(store, &path, after);
+                        let values = try_join!(before_future, after_future);
+                        (path, values)
+                    }
+                }
+            })
+            .buffered(store.concurrency().div(2).max(1)),
```
Comment (yuja) [inline cli/src/diff_util.rs:706]: nit: I prefer infix notation `x / 2`

File: cli/src/diff_util.rs (hunk 1 above)
Comment (martinvonz) [reply cli/src/diff_util.rs]: Good idea.

File: cli/src/diff_util.rs (hunk 2 above)
Comment (martinvonz) [reply cli/src/diff_util.rs]: Heh, so do I. Fixed.

**Pull request merged**